analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -242,8 +242,12 @@ operation.Kind switch
         private static ProgramState SetBranchingConstraints(ControlFlowBranch branch, ProgramState state, IOperation branchValue)
         {
             var constraint = BoolConstraint.From((branch.Source.ConditionKind == ControlFlowConditionKind.WhenTrue) == branch.IsConditionalSuccessor);
-            state = state.SetOperationConstraint(branchValue, constraint);
-            return branchValue.TrackedSymbol() is { } symbol ? state.SetSymbolConstraint(symbol, constraint) : state;
+            state = state.SetOperationConstraint(branchValue, ObjectConstraint.NotNull).SetOperationConstraint(branchValue, constraint);
+            return branchValue.TrackedSymbol() is { } symbol
+                && symbol.GetSymbolType() is { } symbolType
```

Comment:
Can this be null? AFAICT, whatever that comes out of `TrackedSymbol()` will always also have known SymbolType.

I'd name it `type` for simplicity. It doesn't collide with anything around here.
```suggestion
                && symbol.GetSymbolType() is var type
```

Reply:
> AFAICT, whatever that comes out of TrackedSymbol() will always also have known SymbolType

Yes. But I don't mind the extra null check here.

Reply:
The CFG is based on Semantic model in a good shape. We can rely on internal consistency until AD0001 says otherwise => let's make it `var`.

Reply:
I'm not concerned about the semantic model here. It returns [IErrorTypeSymbol](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.ierrortypesymbol?view=roslyn-dotnet-4.3.0) if it has trouble resolving the type. I'm worried about our own `GetSymbolType()`, which may return null. If we enable nullable annotations, we will get a warning here if we use `var`.